plugins/dell-dock/fu-plugin-dell-dock.c

```diff
@@ -201,15 +217,9 @@ fu_plugin_composite_cleanup (FuPlugin *plugin,
 	if (!fu_device_locker_close (locker, error))
 		return FALSE;
 
-	/* if thunderbolt is in the transaction it needs to be activated separately */
-	for (guint i = 0; i < devices->len; i++) {
-		FuDevice *dev = g_ptr_array_index (devices, i);
-		if (g_strcmp0 (fu_device_get_plugin (dev), "thunderbolt") == 0 &&
-		    fu_device_has_flag (dev, FWUPD_DEVICE_FLAG_NEEDS_ACTIVATION)) {
-			if (!fu_device_activate (dev, error))
-				return FALSE;
-			break;
-		}
+	if (needs_activation && dev != NULL) {
```

Comment:
I dont think `needs_activation` can be non-`NULL` but `dev` be `NULL` but doesn't hurt.

Reply:
`needs_activation` is boolean

Reply:
Ah yes, sorry, I mean, `needs_activation` can not be `TRUE` and  `dev` be `NULL` at the same time. But being defensive is probably a better idea anyway!

Reply:
Yeah it's mostly to help prevent if some other code changes this logic in the future and there is a situation that could occur. Thanks!